pkg/usagetracker/tracker_store.go

```diff
@@ -300,16 +341,54 @@ func (shard *tenantShard) trackSeries(series []uint64, now minutes, totalTenantS
 				created = append(created, s)
 				totalTenantSeries.Inc()
 			}
-			pending = pending[1:]
+			missing = missing[1:]
 		}
 		shard.Unlock()
 	}
 
-	// Whatever is pending is rejected (if any)
-	rejected = append(rejected, pending...)
+	// Whatever is still missing is rejected (if any)
+	rejected = append(rejected, missing...)
 	return created, rejected
 }
 
+// processCreatedSeriesEvent creates the series coming from an event from Kafka,
+// i.e. series that were created by a different instance.
+// This does not check the limits, as we prioritize staying consistent across replicas over enforcing limits.
+// Timestamp is not updated if series exists already.
+func (shard *tenantShard) processCreatedSeriesEvent(series []uint64, timestamp minutes, totalTenantSeries *atomic.Uint64) {
+	// missing contains the series that have to be created.
+	// As we advance through series, we reuse the same slice to avoid allocations.
+	missing := series[:0]
+
+	// First try to update the series that already exist, the ones that don't exist are moved to missing.
```

Comment:
[nit] The comment says "update" but we're not doing any update here. We're just looking for missing ones.